src/main/java/de/komoot/photon/nominatim/NominatimConnector.java

```diff
@@ -227,7 +244,7 @@ public void readEntireDatabase(String... countryCodes) {
 
         template.query(SELECT_COLS_PLACEX + " FROM placex " +
                 " WHERE linked_place_id IS NULL AND centroid IS NOT NULL " + andCountryCodeStr +
-                " ORDER BY geometry_sector; ", rs -> {
+                " ORDER BY parent_place_id, geometry_sector; ", rs -> {
```

Comment:
The order should be better ' geometry_sector, parent_place_id'. This ensures that subsequent places are geographically close. This makes a bit of a difference for postgresql because data can be better cached. More surprisingly, it does make a huge difference for ElasticSearch. If the data is not sorted geographically during insert then the database is 20% larger.

Reply:
Thank you for the explanation. We actually did test both ways for Portugal import. Nevertheless, we did not find a huge impact on performance nor in the number of queries made, since the process does not last long enough to reflect that.